pkg/plugin/plugin.go

```diff
@@ -140,8 +142,66 @@ func (pw prefixWriter) Write(buf []byte) (n int, err error) {
 	return len(buf), nil
 }
 
+func getPluginName(archiveName string) string {
+	// if the archive name contains ".app", then it's a macOS app
+	if strings.Contains(archiveName, ".app") {
+		return strings.Split(archiveName, ".app")[0]
+	}
+
+	return strings.Split(archiveName, "_")[0]
+}
+
+func prepareBinary(pluginFilename, pluginPath string) error {
+	pluginName := filepath.Base(pluginPath)
+	binPath := utils.PluginPath(pluginPath, pluginFilename)
+
+	if _, err := os.Stat(binPath); os.IsNotExist(err) {
+		// If the plugin binary does not exist, it means that the plugin is a MacOS .app directory.
```

Comment:
Don't you think that this part should be macos only.

Reply:
I added

	if runtime.GOOS != "darwin" {
			return fmt.Errorf("plugin binary not found at %s", binPath)
		}

bellow